File: Sorting Algorithm/bubblesort.py
# Name : Dakshay Ahuja
# github username : dakshayahuja

# Defining the bubble sort algo:
def bubble_sort(items):
    changes = passes = 0
    last = len(items)
    swapped = True

    while swapped:
        swapped = False
        passes += 1
        for j in range(1, last):
            if items[j - 1] > items[j]:
                items[j], items[j - 1] = items[j - 1], items[j]  # Swap
                changes += 1
                swapped = True
                last = j

    print(items)


print("Welcome to a Bubble Sort Algorithm in Python!")

# Calling the function:
while True:
    print("Enter as many numbers as you want.\nYou can choose between 0 and 9.\nLeave a space between each one")
    numbers = input()
    items = [int(num) for num in numbers.split() if num.isdigit()]
    if items:
        bubble_sort(items)
    break

File: Sorting Algorithm/BubbleSort.jl
# Name : Luiz Fernando Santos
# github username : LFRusso

# Deffining bubble sort algorithm
function bubbleSort(list)
    swapped = true
    compares = 0
    size = length(list)
    while swapped
        swapped = false
        for j =2:size
            if list[j-1]>list[j]
                # Swapping elements positions
                aux = list[j-1]
                list[j-1] = list[j]
                list[j] = aux
                swapped = true
            end
            compares += 1
        end
    end
    return list
end

# Sample input and output
items = [4, 3, 2, 1, 10, 4, 11, 14, 2, 6]
println("Original list: ", items)

bubbleSort(items)
println("Sorted list: ", items)